cou/steps/plan.py

```diff
@@ -117,3 +119,55 @@ async def create_upgrade_group(
             raise
 
     return group_upgrade_plan
+
+
+def determine_upgrade_target(
+    current_os_release: Optional[OpenStackRelease], current_series: Optional[str]
+) -> str:
+    """Determine the target release to upgrade to.
+
+    Inform user if the cloud is already at the highest supporting release of the current series.
+    :param current_os_release: The current minimum OS release in cloud.
+    :type current_os_release: Optional[OpenStackRelease]
+    :param current_series: The current minimum Ubuntu series in cloud.
+    :type current_series: Optional[str]
+    :raises NoTargetError: When cannot find target to upgrade.
+    :return: The target OS release to upgrade the cloud to.
+    :rtype: str
+    """
+    if not current_os_release:
+        raise NoTargetError(
+            "Cannot determine the current OS release in the cloud. "
+            "Is this a valid OpenStack cloud?"
+        )
+
+    supporting_lts_series = ",".join(list(LTS_TO_OS_RELEASE.keys()))
+    if not current_series:
+        raise NoTargetError(
+            "Cannot determine the current Ubuntu series of the cloud series. "
+            f"The supporting series are: {supporting_lts_series}"
+        )
+
+    try:
+        # Check if the release is the "last" supported by the series
+        if str(current_os_release) == LTS_TO_OS_RELEASE[current_series][-1]:
+            print(
+                f"No upgrade available. '{current_os_release}' (the minimum OS release found "
+                f"in the cloud) is the highest release '{current_series}' (the minimum Ubuntu "
+                "series in the cloud) supports."
+            )
+            sys.exit(0)
```

Comment:
To me this is not an good idea hiding sys exit in code. I think we should raise and exception here. Exception should take that string as message and in CLI we should catch it and print the message.

Reply:
I was actually torn between this approach and what you are suggesting here. After re-consideration, yes, I think putting all `sys.exit` in `cli.py` is better.